test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -2129,6 +2132,30 @@ protected NumShards getNumShards(String index) {
         return nodes;
     }
 
+    /**
+     * Asserts that all shards with the same shardId should have document Ids.
+     */
+    public void assertSameDocIdsOnShards() throws Exception {
+        final Map<ShardId, Set<String>> docIdsPerShard = new HashMap<>();
+        final String[] nodeNames = internalCluster().getNodeNames();
+        for (String nodeName : nodeNames) {
+            final IndicesService indexServices = internalCluster().getInstance(IndicesService.class, nodeName);
+            for (IndexService indexService : indexServices) {
+                for (IndexShard shard : indexService) {
+                    try {
+                        final Set<String> docIds = IndexShardTestCase.getShardDocUIDs(shard, true);
+                        if (docIdsPerShard.containsKey(shard.shardId())) {
+                            assertThat(docIds, equalTo(docIdsPerShard.get(shard.shardId())));
```

Comment:
If this fails, we will need more output. Can you add information about which two shard copies diverged?